sdk/program/src/stake_history.rs

```diff
@@ -81,9 +87,83 @@ impl Deref for StakeHistory {
     }
 }
 
+// HANA my stuff below
+
+#[derive(Debug, PartialEq, Eq, Default, Clone)]
+pub struct StakeHistoryData<'a>(Rc<RefCell<&'a mut [u8]>>);
```

Comment:
i dont have any opinion on what the inner type ought to be, i just did this as a placeholder because it works. in my ideal world we could consume the `RefCell` and hold onto the underlying `&[u8]`, which would solve the return type problem, but i dont know if this is in principle possible because the `AccountInfo` is borrowed rather than owned